src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/SchedulerTimestampRepositoryExtensionsImpl.java

```diff
@@ -34,21 +35,26 @@ public boolean updateTimestamp(String schedulerType, Instant timestamp, long sec
         var update = Update.update(TIMESTAMP_FIELD_NAME, timestamp)
             .set(SCHEDULER_TYPE, schedulerType);
 
-        if (collectionIsNotEmpty()) {
+        if (documentAlreadyExists(schedulerType)) {
             UpdateResult result = mongoOperations.updateFirst(query, update, MESH_TIMESTAMP_COLLECTION_NAME);
 
             return updateSuccessful(result);
         } else {
             LOGGER.info("{} collection does not exist or it is empty. Document with timestamp will be created", MESH_TIMESTAMP_COLLECTION_NAME);
             SchedulerTimestamp schedulerTimestamp = new SchedulerTimestamp(schedulerType, timestampService.getCurrentTimestamp());
-            mongoOperations.save(schedulerTimestamp, MESH_TIMESTAMP_COLLECTION_NAME);
+            try {
+                mongoOperations.save(schedulerTimestamp, MESH_TIMESTAMP_COLLECTION_NAME);
+            } catch (MongoWriteException e) {
+                LOGGER.warn("Unable to create new document for scheduler type " + schedulerType + ". Most likely another instance already created the document.", e);
+            }
             return false;
         }
 
     }
 
-    private boolean collectionIsNotEmpty() {
-        return mongoOperations.getCollection(MESH_TIMESTAMP_COLLECTION_NAME).countDocuments() != 0;
+    private boolean documentAlreadyExists(String schedulerType) {
+        var query = query(where(SCHEDULER_TYPE).is(schedulerType));
+        return mongoOperations.count(query, MESH_TIMESTAMP_COLLECTION_NAME) == 1;
```

Comment:
I'd use `> 0` because somehow there might be more documents and this method won't work properly

Reply:
I couldn't decide on this one. There should never be more than one because of the unique key. I'll log an error if > 1 and do the check >= 1 also so it keeps working regardless.